sim/dex.js

```diff
@@ -473,6 +473,22 @@ class ModdedDex {
 			// @ts-ignore
 			return name;
 		}
+
+		let cached = this.effectCache.get(name);
+		if (!cached) {
+			cached = this.getEffectInner(name);
+			this.effectCache.set(name, cached);
+		}
+		return cached;
+	}
+
+	/**
+	* While this function can technically return any kind of effect at
+	* all, that's not a feature TypeScript needs to know about.
+	* @param {string} name
+	* @return {PureEffect}
+	*/
+	getEffectInner(name) {
```

Comment:
I'd normally not split this into two functions (something something call stack height). Why aren't you doing the same caching methods the other `get` functions do?

Reply:
Mainly due to the ugliness of the return type here and all the `// @ts-ignore`. With `effect` at the bottom its pretty easy, but getMove, getAbility etc return Effects, not PureEffects so I need to do more ugly ignoring and it just seemed easier to contain breaking the types into one method. I think I was also influenced by looking at optimizing getRelevantEffects/getRelevantEffectsInner, but thats a different pattern (one adds additional parameters) and in a different class, so I'll follow the more standard pattern here.

On a side note: interesting you're worried about call stack height as opposed to aiming for small, easily inlineable methods (in the past, JS engines would even count comments and whitespace against you WRT inlining, so short methods were king. I'm not sure to what extent that still holds, but I believe its still relevant). Presumably you've been bitten by stack height in the past, judging by the bail out logic in `runEvent` and the super long methods.

Reply:
Yeah, I've had some trouble with stack traces not containing enough information to diagnose bugs in the past. For performance reasons, stack traces are capped at a specific height. That, and I have a general rule of thumb that if you can't think of a better function name than `[...]Inner`, it shouldn't be its own function.